storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOAnnotationsProcessor.java

```diff
@@ -92,38 +98,91 @@ private TypeElement validateAnnotatedClass(@NotNull final Element annotatedEleme
+    protected void validateAnnotatedField(@NotNull final Element annotatedElement) {
+        // we expect here that annotatedElement is Field or Method, annotation requires that via @Target
 
-        final Element enclosingElement = annotatedField.getEnclosingElement();
+        final Element enclosingElement = annotatedElement.getEnclosingElement();
 
-        if (!enclosingElement.getKind().equals(CLASS)) {
+        if (enclosingElement.getKind() != CLASS) {
             throw new ProcessingException(
-                    annotatedField,
-                    "Please apply " + getTypeAnnotationClass().getSimpleName() + " to fields of class: " + annotatedField.getSimpleName()
+                    annotatedElement,
+                    "Please apply " + getColumnAnnotationClass().getSimpleName() + " to fields or methods of class: " + annotatedElement.getSimpleName()
             );
         }
 
         if (enclosingElement.getAnnotation(getTypeAnnotationClass()) == null) {
             throw new ProcessingException(
-                    annotatedField,
+                    annotatedElement,
                     "Please annotate class " + enclosingElement.getSimpleName() + " with " + getTypeAnnotationClass().getSimpleName()
             );
         }
 
-        if (annotatedField.getModifiers().contains(PRIVATE)) {
+        if (annotatedElement.getKind() == FIELD && annotatedElement.getModifiers().contains(PRIVATE)) {
             throw new ProcessingException(
-                    annotatedField,
-                    getColumnAnnotationClass().getSimpleName() + " can not be applied to private field: " + annotatedField.getSimpleName()
+                    annotatedElement,
+                    getColumnAnnotationClass().getSimpleName() + " can not be applied to private field: " + annotatedElement.getSimpleName()
             );
         }
 
-        if (annotatedField.getModifiers().contains(FINAL)) {
+        if (annotatedElement.getKind() == FIELD && annotatedElement.getModifiers().contains(FINAL)) {
             throw new ProcessingException(
-                    annotatedField,
-                    getColumnAnnotationClass().getSimpleName() + " can not be applied to final field: " + annotatedField.getSimpleName()
+                    annotatedElement,
+                    getColumnAnnotationClass().getSimpleName() + " can not be applied to final field: " + annotatedElement.getSimpleName()
+            );
+        }
+
+        if (annotatedElement.getKind() == METHOD && !((ExecutableElement) annotatedElement).getParameters().isEmpty()) {
+            throw new ProcessingException(
+                    annotatedElement,
+                    getColumnAnnotationClass().getSimpleName() + " can not be applied to method with parameters: " + annotatedElement.getSimpleName()
+            );
+        }
+    }
+
+    /**
+     * Checks that element annotated with {@link StorIOCreatorMeta} satisfies all required conditions
```

Comment:
If you don't mind, pls change your comments, we'll eventually update them all I hope :)